source/common/router/router.h

```diff
@@ -132,11 +175,14 @@ class FilterConfig {
                      context.runtime(), context.random(), std::move(shadow_writer),
                      PROTOBUF_GET_WRAPPED_OR_DEFAULT(config, dynamic_stats, true),
                      config.start_child_span(), config.suppress_envoy_headers(),
-                     context.api().timeSource(), context.httpContext()) {
+                     config.strict_check_headers(), context.api().timeSource(),
+                     context.httpContext()) {
     for (const auto& upstream_log : config.upstream_log()) {
       upstream_logs_.push_back(AccessLog::AccessLogFactory::fromProto(upstream_log, context));
     }
   }
+  using HeaderVector = std::vector<Http::LowerCaseString>;
+  using StrictCheckHeadersPtr = std::unique_ptr<HeaderVector>;
```

Comment:
since this is just a unique ptr to HeaderVector, i'd probably call it just `HeaderVectorPtr`